iree/compiler/InputConversion/MHLO/ConvertMHLOToLinalgExt.cpp

```diff
@@ -161,25 +167,55 @@ struct SortOpConversion : public OpConversionPattern<mhlo::SortOp> {
   using OpConversionPattern<mhlo::SortOp>::OpConversionPattern;
 
   LogicalResult matchAndRewrite(
-      mhlo::SortOp mhloSortOp, OpAdaptor adaptor,
+      mhlo::SortOp op, OpAdaptor adaptor,
       ConversionPatternRewriter &rewriter) const final {
+    Location loc = op.getLoc();
+    llvm::SmallVector<Value> inputs;
+    for (auto input : adaptor.getOperands()) {
```

Comment:
This pattern has a type converter that we should be able to add a pattern too. I only had it handle scalars because at the time, unsigned function signatures didn't work when converting from linalg at all. You should be able to upgrade it and use it here. I guess the reason the inputs aren't getting converted is because function signature conversion is done as part of the subsequent mhlo pass. Ugh what a mess. These two passes should really be combined, I think.

Reply:
Managed to rework using the type converter.